thirdparty/faiss/faiss/invlists/OnDiskInvertedLists.cpp

```diff
@@ -674,10 +674,33 @@ void OnDiskInvertedLists::set_all_lists_sizes(const size_t* sizes) {
     for (size_t i = 0; i < nlist; i++) {
         lists[i].offset = ofs;
         lists[i].capacity = lists[i].size = sizes[i];
-        ofs += sizes[i] * (sizeof(idx_t) + code_size);
+        if (this->with_norm) {
+            ofs += sizes[i] * (sizeof(idx_t) + code_size + sizeof(float));
+        } else {
+            ofs += sizes[i] * (sizeof(idx_t) + code_size);
+        }
+    }
+}
+
+const float* OnDiskInvertedLists::get_code_norms(
+        size_t list_no,
+        size_t /*offset*/) const {
+    if (with_norm) {
+        if (lists[list_no].offset == INVALID_OFFSET) {
+            return nullptr;
+        }
+        return (const float*)(ptr + lists[list_no].offset +
+                            (code_size + sizeof(idx_t)) * lists[list_no].capacity);
+    } else {
+        return nullptr;
     }
 }
 
+void OnDiskInvertedLists::release_code_norms(size_t list_no, const float* codes)
```

Comment:
why is it needed to override this method in the provided way?

Reply:
Just to clarify the question. It seems that the base class has the same implementation as the added one

Reply:
I thought the behavior in release_code_norms is based on the storage method of InvertedList, so this function is overrided, and this function do nothing like the basic class.

Reply:
Then I would eliminate it, because it does nothing.
Or I can lgtm the PR, just let me know. :)